src/lifecycle/properties-init.js

```diff
@@ -4,147 +4,90 @@ import data from '../util/data';
 import emit from '../api/emit';
 import empty from '../util/empty';
 
-const { removeAttribute, setAttribute } = window.Element.prototype;
-
-function getData (elem, name) {
-  return data(elem, `api/property/${name}`);
+function getDefaultValue (elem, name, opts) {
+  return typeof opts.default === 'function' ? opts.default(elem, { name }) : opts.default;
 }
 
-function getDataForAttribute (elem, name) {
-  return getData(elem, getData(elem, name).linkedProperty);
+function getInitialValue (elem, name, opts) {
+  typeof opts.initial === 'function' ? opts.initial(elem, { name }) : opts.initial;
```

Comment:
Added some tests. It required a tiny bit of refactoring in the test to accommodate `initial` tests, but happy they got written. Totally my oversight they didn't get in initially (no pun intended).